.scripts/pm_apt_install.sh

```diff
@@ -4,38 +4,97 @@ IFS=$'\n\t'
+                    info "Adding/updating sources for ${APPNAME}"
+                    if [[ -f ${SOURCE_FILE} ]]; then
+                        rm "${SOURCE_FILE}"
+                    fi
+                    echo "${SOURCE_REPO}" | tee "${SOURCE_FILE}" > /dev/null
+                    UPDATE_APT="true"
+                fi
+            elif [[ ${SOURCE_REPO} == ppa:* ]]; then
+                info "Adding/updating sources for ${APPNAME}"
+                add-apt-repository -y "${SOURCE_REPO}" > /dev/null
+                UPDATE_APT="true"
+            else
+                error "Source Repo is an invalid format. Must start be 'deb' or 'ppa:'!"
+                error "${SOURCE_REPO}"
+                return 1
             fi
-            echo "${SOURCE_REPO}" | tee "${SOURCE_FILE}" > /dev/null
         fi
 
         local SOURCE_KEY
-        SOURCE_KEY=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt_key" || true)
+        SOURCE_KEY=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt.${DETECTED_DISTRO}.${DETECTED_CODENAME}.key" || true)
+        if [[ ${SOURCE_KEY} == "" ]]; then
+            SOURCE_KEY=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt.${DETECTED_DISTRO}.key" || true)
+        fi
+        if [[ ${SOURCE_KEY} == "" ]]; then
+            SOURCE_KEY=$(run_script 'yml_get' "${APPNAME}" "services.${FILENAME}.labels[com.appstarter.appinstall].apt.general.key" || true)
+        fi
         if [[ ${SOURCE_KEY} != "" ]]; then
-            info "Adding/updating source key for ${APPNAME}"
-            gpg --list-keys "${SOURCE_KEY}" > /dev/null 2>&1 || apt-key adv --keyserver keyserver.ubuntu.com --recv-keys "${SOURCE_KEY}" > /dev/null 2>&1
+            if [[ ${SOURCE_KEY} == http* ]]; then
+                info "Adding/updating source key for ${APPNAME}"
+                wget -qO - "${SOURCE_KEY}" | apt-key add -  > /dev/null 2>&1 || error "Unable to add key for ${APPNAME}: ${SOURCE_KEY}"
+                UPDATE_APT="true"
+            else
+                if ! gpg --list-keys "${SOURCE_KEY}" > /dev/null 2>&1; then
+                    info "Adding source key for ${APPNAME}"
+                    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys "${SOURCE_KEY}" > /dev/null 2>&1
+                    UPDATE_APT="true"
+                fi
+            fi
+        fi
+
+        if [[ ${UPDATE_APT:-} == "true" ]]; then
+            run_script 'package_manager_run' repos
         fi
 
-        run_script 'package_manager_run' repos
-        apt-get -y install "${APP_PACKAGE}" > /dev/null 2>&1 || error "Failed to install/update ${APPNAME} from apt."
-        run_script 'package_manager_run' clean
+        PACKAGE_VERSION_INSTALLED=$(sudo apt-cache policy "${APP_PACKAGE}" | grep "Installed:" | awk '{gsub("Installed:", ""); gsub(" ", ""); print}')
```

Comment:
`sudo apt-cache policy "${APP_PACKAGE}" | grep "Installed:" | awk '{gsub("Installed:", ""); gsub(" ", ""); print}'` could be `sudo apt-cache policy "${APP_PACKAGE}" | grep -Po "Installed: \K.*"`